mailpoet/lib/Entities/ScheduledTaskEntity.php

```diff
@@ -211,6 +212,19 @@ public function getSubscribers(): Collection {
     return $this->subscribers;
   }
 
+  /**
+   * @param int $processed ScheduledTaskSubscriberEntity::PROCESSED_* constant
+   * @return SubscriberEntity[]
+   */
+  public function getSubscribersByProcessed(int $processed): array {
+    $criteria = Criteria::create()
+      ->where(Criteria::expr()->eq('processed', $processed));
+    $subscribers = $this->subscribers->matching($criteria)->map(function (ScheduledTaskSubscriberEntity $taskSubscriber): ?SubscriberEntity {
+      return $taskSubscriber->getSubscriber();
+    });
+    return array_filter($subscribers->toArray());
```

Comment:
Why `array_filter()`?

Reply:
Because `ScheduledTaskSubscriberEntity::getSubscriber()` can return `null` or `SubscriberEntity` and PHPStan reports an error here.

Reply:
gotcha.